src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -221,27 +306,138 @@ static ucs_status_t uct_cuda_copy_mem_alloc(uct_md_h md, size_t *length_p,
         return UCS_ERR_NO_DEVICE;
     }
 
+    alloc_handle = ucs_malloc(sizeof(*alloc_handle),
+                              "uct_cuda_copy_alloc_handle_t");
+    if (NULL == alloc_handle) {
+        ucs_error("failed to allocate memory for uct_cuda_copy_alloc_handle_t");
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    alloc_handle->length = *length_p;
```

Comment:
uct_cuda_copy_mem_alloc_fabric may increase the length due to alignment. so if alloc is successful, need to update *length_p accordingly.